src/shared/components/header/header.component.ts

```diff
@@ -7,6 +9,12 @@ import { ChangeDetectionStrategy, Component } from '@angular/core';
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class HeaderComponent {
+  constructor(public dialog: Dialog) {}
+
+  openDialog() {
+    this.dialog.open(DynamicFormComponent);
+  }
```

Comment:
Your component shouldn't say that it's dynamic or not. Change the name of component and say what this form is about.